docs/components/renderer.rst

```diff
@@ -6,5 +6,51 @@
 Renderer
 ========
 
-..  warning::
-    This needs to be documented
+A Renderer transforms the :abbreviation:`AST (abstract syntax tree)` into a
+desired output format.
+
+The most common renderers handle each document separately. For example the
+:php:class:`\phpDocumentor\Guides\Renderer\HtmlRenderer` renders the AST into
+HTML.
+
+Each renderer must implement
+:php:interface:`\phpDocumentor\Guides\Renderer\TypeRenderer`. Basic document
+type renderers like the HTML or Latex renderer can extend
+the :php:class:`\phpDocumentor\Guides\Renderer\BaseTypeRenderer`.
+
+All renderers must be registered in the ContainerConfigurator of the extension
+with the tag :php:`'phpdoc.renderer.typerenderer'` and additional format information.
+
+Example: a plaintext renderer
+=============================
+
+Create a class called :php:`PlaintextRenderer` which extends
+:php:class:`\phpDocumentor\Guides\Renderer\BaseTypeRenderer` and is
+empty otherwise.
```

Comment:
I suppose the reason for an empty renderer is that the BaseTypeRenderer already renders als plaintext? Maybe some note about this like "and is empty otherwise (because the BaseTypeRenderer will provide a raw unaltered rendering of the AST)"?